### PR TITLE
Feature/7648 family/7840 preferred person and certificate

### PR DESCRIPTION
**Pull request by nickguendling**
## Description
Adds a preferred person flag and the most relevant health certificate to the `HealthCertifiedPerson`.

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-7840

### CODE REVIEW

**Review by KaiTeuber — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/__test__/HealthCertificateArrayMostRelevantTests.swift [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+			olderSeriesCompletingVaccinationCertificate,
+			mostRecentOtherVaccinationCertificate,
+			olderOtherVaccinationCertificate,
+			mostRecentOutdatedRecoveryCertificate,
+			olderOutdatedRecoveryCertificate,
+			mostRecentOutdatedPCRTest,
+			olderOutdatedPCRTest,
+			mostRecentOutdatedAntigenTest,
+			olderOutdatedAntigenTest
+		].shuffled()
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/__test__/HealthCertificateArrayMostRelevantTests.swift:59]: Finally a usecase for `shuffled()` 😁

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/HealthCertificate.swift [hunk 2]
```diff
@@ -87,6 +93,17 @@ struct HealthCertificate: Codable, Equatable, Comparable {
 	}
 
 	var type: CertificateType {
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/HealthCertificate.swift:95]: Whats the reason for having type? Cant we just only use entry? If you read the entry and you don't need the associated value, you can omit it.

**Review by ArturFriesen — CHANGES_REQUESTED** [review]

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/HealthCertificate.swift (hunk 2 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/HealthCertificate.swift:95]: With associated value I couldn't just write e.g. `$0.entry == .test`, as the default `Equatable` implementation always requires an associated value to equate to. And the `if case .test = $0.entry` variant is very inconvenient for `.filter`ing etc.

**Review by naveeddotio — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/HealthCertificate.swift (hunk 2 above)
Comment (ArturFriesen) [reply src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/HealthCertificate.swift:95]: Ok, I unterstand!

**Review by ArturFriesen — APPROVED** [review]

**Pull request merged**